### PR TITLE
Enabling unprovisionned mdns for efr32 boards

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Problem
* enabling  unprovisionned mdns for efr32 boards

#### Change overview
* Increased stack to support unprovisionned mdns
* Enabled unprovisionned mdns

#### Testing
* Manual testing with discovery app to validate broadcasting
* Validation with `./chip-tool discover commissionables` to validate broadcasting

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: src/platform/EFR32/CHIPDevicePlatformConfig.h [hunk 1]
```diff
@@ -36,6 +36,8 @@
 #define CHIP_DEVICE_CONFIG_ENABLE_THREAD 1
 #define CHIP_DEVICE_CONFIG_ENABLE_MDNS 1
 #define CHIP_DEVICE_CONFIG_ENABLE_THREAD_SRP_CLIENT 1
+#define CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY 1
+#define CHIP_DEVICE_CONFIG_ENABLE_UNPROVISIONED_MDNS 1
```
Comment (tcarmelveilleux) [inline src/platform/EFR32/CHIPDevicePlatformConfig.h:40]: Why is this called MDNS when it's DNS-SD ? We should rename this `CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONABLE_DISCOVERY`. On thread devices, it should use SRP.

File: examples/lock-app/efr32/src/AppTask.cpp [hunk 2]
```diff
@@ -50,7 +50,7 @@
 
 #define FACTORY_RESET_TRIGGER_TIMEOUT 3000
 #define FACTORY_RESET_CANCEL_WINDOW_TIMEOUT 3000
-#define APP_TASK_STACK_SIZE (1536)
+#define APP_TASK_STACK_SIZE (4096)
```
Comment (tcarmelveilleux) [inline examples/lock-app/efr32/src/AppTask.cpp:53]: What is the reason for the growth? What was the straw that broke the camel's back? This is a big increase.

Comment (mkardous-silabs) [reply examples/lock-app/efr32/src/AppTask.cpp:53]: Enabling unprovisioned_mdns causes an increased stack usage at launch of the app task for all the necessary configurations for the mdns server.  We are increasing it for the time being until we optimize it.

File: src/platform/EFR32/CHIPDevicePlatformConfig.h (hunk 1 above)
Comment (mkardous-silabs) [reply src/platform/EFR32/CHIPDevicePlatformConfig.h]: done

**Review by msandstedt — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

File: examples/lock-app/efr32/src/AppTask.cpp (hunk 2 above)
Comment (jmartinez-silabs) [reply examples/lock-app/efr32/src/AppTask.cpp:53]: It has to do with a case session now being done in `InitServer();` in the app task for the unprovisionned Broadcast. I looked it up with @mkardous-silabs But he can probably provide more details.

**Pull request merged**